syntax = "proto3";

import "svg.proto";

package map_renderer_proto;

message RenderSettings {
	double width = 1;
	double height = 2;
	double padding = 3;
	double stop_radius = 4;
	double line_width = 5;
	int32 bus_label_font_size = 6;
	svg_proto.Point bus_label_offset = 7;
	int32 stop_label_font_size = 8;
	svg_proto.Point stop_label_offset = 9;
	svg_proto.Color underlayer_color = 10;
	double underlayer_width = 11;
	repeated svg_proto.Color color_palette = 12;
}

message MapRenderer {
	RenderSettings settings = 1;
	repeated svg_proto.Polyline polyline = 2;
	repeated svg_proto.Circle circle = 3;
}